docs/src/content/docs/basic-concepts.mdx

```diff
@@ -49,9 +50,28 @@ const AIGreeting = gsx.Component<GreetingProps, string>(async ({ name }) => {
 });
 ```
 
+Components can return structured objects containing other components. The pattern also works with array operations like `map` to process items in parallel.
```

Comment:
"Components can return arrays and structured objects containing other ocmponents"

Reply:
updated